### PR TITLE
DBZ-927 : Exception while parsing table schema with invalid default value for timestamp field

### PR DESCRIPTION
**Pull request by deepakbarr**
https://issues.jboss.org/browse/DBZ-927

Summary of Changes :
1. Changed the regex expression for matching the zero date and datetime values.
2. Added test case for various date values.
3. Modified test case of testing datetime values containing zeros

### CODE REVIEW

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/AbstractMysqlDefaultValueTest.java [hunk 1]
```diff
@@ -339,7 +339,10 @@ public void parseTimeDefaultValue() {
                 "  I datetime(3) NOT NULL DEFAULT '0000-00-00 00:00:00.000'," +
                 "  J datetime NOT NULL DEFAULT '2018-06-26 12:34:56'," +
                 "  K datetime(3) NOT NULL DEFAULT '2018-06-26 12:34:56.000'," +
-                "  L datetime(2) NOT NULL DEFAULT '2018-06-26 12:34:56.78'" +
+                "  L datetime(2) NOT NULL DEFAULT '2018-06-26 12:34:56.78'," +
+                "  M datetime NOT NULL DEFAULT '2000-01-00 00:00:00'," +
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/AbstractMysqlDefaultValueTest.java]: So to make sure I get this right: all these values result in a default value of the epoch, as they are invalid essentially?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java [hunk 2]
```diff
@@ -33,9 +33,10 @@
 @Immutable
 public class MySqlDefaultValuePreConverter  {
 
-    private static final Pattern ALL_ZERO_TIMESTAMP = Pattern.compile("0000-00-00 00:00:00(\\.\\d{1,6})?");
 
-    private static final String ALL_ZERO_DATE = "0000-00-00";
+    private static final Pattern ALL_ZERO_TIMESTAMP = Pattern.compile("(\\d{4}-\\d{2}-00|\\d{4}-00-\\d{2}|0000-\\d{2}-\\d{2}) (00:00:00(\\.\\d{1,6})?)");
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java]: Seems these patterns should be renamed now, e.g. something like `EPOCH_EQUIVALENT_TIMESTAMP`?

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/AbstractMysqlDefaultValueTest.java (hunk 1 above)
Comment (deepakbarr) [reply debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/AbstractMysqlDefaultValueTest.java]: Yes. The basic idea is the timestamp need not be "ALL" zeros. Sometimes, only month part can be zero OR only day part can be zero etc. All of them are invalid. So, If the column is optional, we use null. If the column is non optional, we use EPOCH timestamp.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java (hunk 2 above)
Comment (deepakbarr) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValuePreConverter.java]: I have renamed them.

**Comment by deepakbarr** [comment]
I have addressed the review comments.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks a lot, @deepakbarr!

**Pull request closed without merging**